code_generation/code_gen.py

```diff
@@ -117,17 +117,45 @@ def render_dataset_class_maps(self, template_path: Path, data_path: Path, output
 
         self.render_template(template_path=template_path, output_path=output_path, all_map=all_map)
 
+    def render_metadata_types(self, template_path: Path, data_path: Path, output_path: Path):
+        with open(data_path) as data_file:
+            json_data = data_file.read()
+        dataset_meta_data: List[DatasetMapData] = AllDatasetMapData.schema().loads(json_data).all_datasets
+
+        dataset_types = []
+        components = []
+        for dataset in dataset_meta_data:
+            if dataset.is_template:
+                prefixes = ["sym_", "asym_"]
+            else:
+                prefixes = [""]
+            for prefix in prefixes:
+                dataset_types.append(f"{prefix}{dataset.name}")
+
+                if dataset.name == "input":
+                    for component in dataset.components:
+                        components.append(component.names)
+
+        components = [name for sublist in components for name in sublist]
+
+        self.render_template(
+            template_path=template_path, output_path=output_path, dataset_types=dataset_types, components=components
+        )
+
     def code_gen(self):
         render_funcs = {
             "attribute_classes": self.render_attribute_classes,
             "dataset_class_maps": self.render_dataset_class_maps,
+            "metadata_enums": self.render_metadata_types,
         }
 
         # render attribute classes
         for template_name, render_func in render_funcs.items():
             for template_path in TEMPLATE_DIR.rglob(f"{template_name}.*.jinja"):
                 output_suffix = template_path.with_suffix("").suffix
                 output_dir = template_path.parent.relative_to(TEMPLATE_DIR)
+                if template_name == "metadata_enums":
+                    template_name = "dataset_class_maps"  # To use existing data.
                 for data_path in DATA_DIR.glob(f"{template_name}/*.json"):
```

Comment:
```suggestion
                data_name = "dataset_class_maps" if template_name == "metadata_enums" else template_name
                for data_path in DATA_DIR.glob(f"{data_name}/*.json"):
```
Sort of a nitpick here for a script. But creating new variable is always better than modifying existing one.